packages/node_modules/pouchdb-abstract-mapreduce/src/taskqueue.js

```diff
@@ -3,18 +3,16 @@
  * callbacks will eventually fire (once).
  */
 
-
 class TaskQueue {
   constructor() {
-    this.promise = new Promise(function (fulfill) {fulfill(); });
+    this.promise = Promise.resolve();
   }
 
   add(promiseFactory) {
-    this.promise = this.promise.catch(function () {
+    this.promise = this.promise
       // just recover
-    }).then(function () {
-      return promiseFactory();
-    });
+      .catch(function () { })
+      .then(promiseFactory);
```

Comment:
Technically you've changed the behaviour here because any return value from the `promise` will be passed to `promiseFactory` as a parameter. I'm too lazy to work out if this will break anything, but to be safe let's change it to...

```suggestion
      .then(() => promiseFactory());
```

Reply:
Sure, I'll provide a follow up pr for this.